x-pack/libbeat/licenser/es_callback.go

```diff
@@ -0,0 +1,40 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package licenser
+
+import (
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/common/atomic"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/outputs/elasticsearch"
+)
+
+// Enforce setups the corresponding callbacks in libbeat to verify the license on the
+// remote elasticsearch cluster.
+func Enforce(log *logp.Logger, checks ...CheckFunc) {
+	validLicense := atomic.MakeBool(false)
```

Comment:
What requires the use of the atomic bool? It appears that `validLicense` could be scoped to the callback function.

Reply:
I was worried that invoke multiple times, but then again might be something we want.